mockmaker/bytebuddy/main/java/org/mockito/internal/creation/bytebuddy/MockMethodInterceptor.java

```diff
@@ -29,46 +29,6 @@ public MockMethodInterceptor(InternalMockHandler handler, MockCreationSettings m
         serializationSupport = new ByteBuddyCrossClassLoaderSerializationSupport();
     }
 
-    @RuntimeType
-    @BindingPriority(BindingPriority.DEFAULT * 3)
-    public Object interceptSuperCallable(@This Object mock,
-                                         @Origin(cache = true) Method invokedMethod,
-                                         @AllArguments Object[] arguments,
-                                         @SuperCall(serializableProxy = true) Callable<?> superCall) throws Throwable {
-        return doIntercept(
-                mock,
-                invokedMethod,
-                arguments,
-                new InterceptedInvocation.SuperMethod.FromCallable(superCall)
-        );
-    }
-
-    @RuntimeType
-    @BindingPriority(BindingPriority.DEFAULT * 2)
-    public Object interceptDefaultCallable(@This Object mock,
-                                           @Origin(cache = true) Method invokedMethod,
-                                           @AllArguments Object[] arguments,
-                                           @DefaultCall(serializableProxy = true) Callable<?> superCall) throws Throwable {
-        return doIntercept(
-                mock,
-                invokedMethod,
-                arguments,
-                new InterceptedInvocation.SuperMethod.FromCallable(superCall)
-        );
-    }
-
-    @RuntimeType
-    public Object interceptAbstract(@This Object mock,
-                                    @Origin(cache = true) Method invokedMethod,
-                                    @AllArguments Object[] arguments) throws Throwable {
-        return doIntercept(
-                mock,
-                invokedMethod,
-                arguments,
-                InterceptedInvocation.SuperMethod.IsIllegal.INSTANCE
-        );
-    }
-
```

Comment:
Byte Buddy delegates to the dispatcher. The dispatcher can then decide what to do with an invocation based on of the field is initialized.